packages/next-codemod/bin/upgrade.ts

```diff
@@ -651,3 +661,119 @@ function writeOverridesField(
     }
   }
 }
+
+function warnPeerDependencies(
+  appPackageJson: any,
+  versionMapping: Record<string, { version: string; required: boolean }>
+) {
+  const versionMappingKeys = Object.keys(versionMapping)
+  const allDependenciesToInstall = {
+    ...appPackageJson.dependencies,
+    ...appPackageJson.devDependencies,
+  }
+  const peerDepsOutOfRange = new Map<
+    string,
+    { current: string; expected: string; from: string }
+  >()
+
+  if ('peerDependencies' in appPackageJson) {
+    const peerDeps = appPackageJson.peerDependencies
+    const peerDepsKeys = Object.keys(peerDeps)
+    const commonKeys = versionMappingKeys.filter((versionMappingKey) =>
+      peerDepsKeys.includes(versionMappingKey)
+    )
+
+    if (commonKeys.length > 0) {
+      for (const key of commonKeys) {
+        const peerDepVersion = peerDeps[key]
+        const { version } = versionMapping[key]
+
+        if (
+          !semverSatisfies(version, peerDepVersion, {
+            includePrerelease: true,
+          })
+        ) {
+          peerDepsOutOfRange.set(key, {
+            current: version,
+            expected: peerDepVersion,
+            from: key,
+          })
+        }
+      }
+    }
+  }
+
+  for (const dependency of Object.keys(allDependenciesToInstall)) {
+    // use readFile to ensure it reads after the install is complete
+    const pkgJson = JSON.parse(
+      fs.readFileSync(
+        path.join(cwd, 'node_modules', dependency, 'package.json'),
+        'utf8'
+      )
+    )
+
+    if ('peerDependencies' in pkgJson) {
+      const peerDeps = pkgJson.peerDependencies
+      const peerDepsKeys = Object.keys(peerDeps)
+      const commonKeys = versionMappingKeys.filter((versionMappingKey) =>
+        peerDepsKeys.includes(versionMappingKey)
+      )
+
+      if (commonKeys.length > 0) {
```

Comment:
This is redundant. The loop doesn't do anything if `commonKeys.length === 0` so you can just remove this `if`.